pennylane/_qubit_device.py

```diff
@@ -269,6 +269,20 @@ def statistics(self, observables):
             elif obs.return_type is Probability:
                 results.append(self.probability(wires=obs.wires))
 
+            elif obs.return_type is State:
+                if len(obs.wires) != self.num_wires:
+                    raise QuantumFunctionError("The state must be returned over all wires")
+
+                try:
+                    state = self.state
+                except AttributeError:
+                    state = None
+
+                if state is None:
+                    raise AttributeError("The state is not available in the current device")
```

Comment:
To avoid the awkwardness of doing it this way, you could do someting like `getattr(self, "state", None`)?